tests/integration/test_exporter.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import time
+
+import pytest
+import urllib3
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+MYSQL_APP_NAME = "mysql"
+MYSQL_ROUTER_APP_NAME = "mysql-router"
+APPLICATION_APP_NAME = "mysql-test-app"
+GRAFANA_AGENT_APP_NAME = "grafana-agent"
+SLOW_TIMEOUT = 25 * 60
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_exporter_endpoint(ops_test: OpsTest, mysql_router_charm_series: str) -> None:
+    """Test that exporter endpoint is functional."""
+    http = urllib3.PoolManager()
+
+    # Build and deploy applications
+    mysqlrouter_charm = await ops_test.build_charm(".")
+
+    logger.info("Deploying all the applications")
+
+    # deploy mysqlrouter with num_units=None since it's a subordinate charm
+    # and will be installed with the related consumer application
+    applications = await asyncio.gather(
+        ops_test.model.deploy(
+            "mysql",
```

Comment:
```suggestion
            MYSQL_APP_NAME,
```

Reply:
addressed in 8ef349b3570ea7f66f44e4f7c9625fc97c5ea690